### PR TITLE
[FSEC-1576] Add encryption rule

### PR DESCRIPTION
**Pull request by eeshaan**
# Description

Adds a rule to check for encryption of DynamoDB tables. Currently not enabled in "Foundation Secure Security Compliance Benchmarking" as there's no way to disable default encryption. 

Resolves FSEC-1576

## References
- https://zendesk.atlassian.net/browse/FSEC-1576

## Type of change

Select the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/providers/aws/rules/findings/dynamodb-no-default-encryption.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "description": "Table without Server-Side Encryption at Rest Enabled",
```
Comment (awagneratzendesk) [inline ScoutSuite/providers/aws/rules/findings/dynamodb-no-default-encryption.json]: I'm thinking we want to change the wording of this. Based on what you told us last week, DDB is always encrypted no matter what. If this is not true, the data is encrypted by the AWS owned key that is not linked to our account. Maybe we can clarify that in the description or rationale?

**Comment by hannahgreene** [comment]
Is there any case where this test returns positive flagged resources. Didn't we determine that encryption is always enabled? The backups rule we definitely can add, though

**Review by hannahgreene — COMMENTED** [review]
I think we should close this in favor of the backups rule

**Comment by eeshaan** [comment]
I agree with the points made; just wasn't sure if we wanted the rule to still be available in `findings/` even if we didn't want it in our own ruleset for now. Will close if needed.

**Comment by hannahgreene** [comment]
> I agree with the points made; just wasn't sure if we wanted the rule to still be available in `findings/` even if we didn't want it in our own ruleset for now. Will close if needed.

I'm good to leave it open if it is useful in other contexts, but I'm not sure if it'll be able to actually flag anything

**Comment by eeshaan** [comment]
> not sure if it'll be able to actually flag anything

Yeah, probably not. Maybe we can leave it open for now and modify it in the future to check for the encryption type instead of just enabled/disabled like @awagneratzendesk was suggesting.
image

**Comment by awagneratzendesk** [comment]
Looking at the Cloudformation spec https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-ssespecification.html#cfn-dynamodb-table-ssespecification-sseenabled - if sse_enabled = false, we're using AWS-owned CMKs. If sse_enabled = true, we're using our own KMS keys.

So I think we're looking at the right field to check for the different types of encryption keys.

**Review by gmathes — APPROVED** [review]

**Pull request merged**